econplayground/api/tests/test_views.py

```diff
@@ -161,14 +161,35 @@ def test_create(self):
         })
         self.assertEqual(response.status_code, 201)
         self.assertEqual(Submission.objects.count(), 1)
+        self.assertEqual(Submission.objects.first().choice, 3)
+
+    def test_create_dup_fail(self):
+        response = self.client.post('/api/submissions/', {
+            'graph': self.g.pk,
+            'choice': 3,
+        })
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(Submission.objects.count(), 1)
+
+        self.client.post('/api/submissions/', {
+            'graph': self.g.pk,
+            'choice': 4,
+        })
+
+        # Somehow, the ValidationError is getting swallowed and not
+        # setting the status_code to 400.
```

Comment:
Note to self: the ValidationError is properly displayed in the djangorestframework admin UI, but not in the tests, or when actually tested in javascript with XHR. In practice, this isn't a big deal - the database will throw an IntegrityError when it detects a duplicate, and all this changes is where and how we handle which error.